src/main/java/org/truffleruby/core/module/ModuleFields.java

```diff
@@ -581,7 +581,7 @@ public String toString() {
     }
 
     private void newConstantsVersion() {
-        constantsUnmodifiedAssumption.invalidate();
+        constantsUnmodifiedAssumption.invalidate(givenBaseName + " constants were modified");
```

Comment:
The assumptions are used in more places that `newConstantsVersion` and `newMethodsVersion`. But I think those are the only places where they are `invalidate`d?

Reply:
Yeah, looks that way. All the other assumptions in TruffleRuby appear to be created with string constants so don't need changing to fix that, though they also don't give invalidation reasons so might be worth tidying up in a future PR.

Reply:
yes, we can add invalidation reasons everywhere now. If the assumption's name is meaningful anyway, then this won't be necessary, though.

Reply:
What about having just the `givenBaseName` as the reason? People will see the stack trace and can work out what the reason was beyond the name, and it's one less allocation (I think we get a lot of invalidations during startup).

Reply:
I think we should add a reason to `newConstantsVersion` which would say what caused the invalidation.
But this doesn't need to be in this PR.
Since we already have the `constants are unmodified` in the Assumption name, it seems reundant to add it here.

Reply:
ok, I will remove the `... are umodified` from the invalidation (just the `givenBaseName`) there.